cimgraph/data_profile/cimhub_2023/cimhub_2023.py

```diff
@@ -0,0 +1,18692 @@
+    mRID is superseded by Identity.identifier, which is typed to be a UUID.
+    '''
+    identifier: Optional[UUID] = field(
+        default = None,
+        metadata = {
+            'type': 'Attribute',
+            'minOccurs': '1',
+            'maxOccurs': '1'
+        })
+
+    # Backwards support for objects created with mRID
+    def __post_init__(self) -> None:
+        if 'mRID' in self.__dataclass_fields__:
+            if self.mRID is not None:
+                self.uuid(mRID = self.mRID)
+
+    # Override python string for printing with JSON representation
+    def __str__(self) -> str:
+        # Create JSON-LD dump with repr and all attributes
+        dump = dict(json.loads(self.__repr__()) | self.__dict__)
+        del dump['__uuid__']
+        attribute_list = list(self.__dataclass_fields__.keys())
+        for attribute in attribute_list:
+            # Delete attributes from print that are empty
+            if dump[attribute] is None or dump[attribute] == []:
+                del dump[attribute]
+            # If a dataclass, replace with custom repr
+            elif is_dataclass(dump[attribute]):
+                dump[attribute] = dump[attribute].__repr__()
+            elif type[dump[attribute]] != str:
+                # Reformat all attributes as string for JSON
+                dump[attribute] = str(dump[attribute])
+        # Fix python ' vs JSON "
+        dump = json.dumps(dump)
+        dump = str(dump).replace('\\\"','\"' )
+        dump = str(dump).replace('\"[','[' )
+        dump = str(dump).replace(']\"',']' )
+        dump = str(dump).replace('\"{','{' )
+        dump = str(dump).replace('}\"','}' )
+        return dump
+
+    # Override python __repr__ method with JSON-LD representation
+    # This is needed to avoid infinite loops in object previews
+    def __repr__(self) -> str:
+        return json.dumps({'@id': f'{str(self.identifier)}', '@type': f'{self.__class__.__name__}'})
+
+    # Add indentation of json for pretty print
+    def pprint(self) -> None:
+        print(json.dumps(json.loads(self.__str__()), indent=4))
+
+    # Create UUID from inconsistent mRIDs
+    def uuid(self, mRID:str = None, uri:str = None, name:str = None) -> UUID:
+        seed = ''
+        invalid_mrid = True
+        self.__uuid__ = self.__uuid_meta__()
+        # If URI is specified, try creating from UUID from URI
+        if uri is not None:
+            # Handle inconsistent capitalization / underscores
+            if uri.strip('_') != uri:
+                self.__uuid__.uri_has_underscore = True
+            if uri.lower() != uri:
+                self.__uuid__.uri_is_capitalized = True
+            try:
+                self.identifier = UUID(uri.strip('_').lower())
+                invalid_mrid = False
+            except:
+                seed = seed + uri
+                _log.warning(f'URI {uri} not a valid UUID, generating new UUID')
+        if mRID is not None and str(self.identifier) != self.mRID:
```

Comment:
```suggestion
        if mRID is not None:
            if str(self.identifier) != mRID:
                # Handle inconsistent capitalization / underscores
                if mRID.strip('_') != mRID:
                    self.__uuid__.mrid_has_underscore = True
                    if uri is None:
                        self.__uuid__.uri_has_underscore = True
                if mRID.lower() != mRID:
                    self.__uuid__.mrid_is_capitalized = True
                    if uri is None:
                        self.__uuid__.uri_is_capitalized = True
                try:
                    self.identifier = UUID(mRID.strip('_').lower())
                    invalid_mrid = False
                except:
                    seed = seed + mRID
                    _log.warning(f'mRID {mRID} not a valid UUID, generating new UUID')
            else:
                invalid_mrid = False
            if 'mRID' in self.__dataclass_fields__:
                self.mRID = mRID
```

Reply:
The suggested change in here fixes an issue where the user/post_init calls uuid(mRID=) such that mRID = str(identifier) and invalid_mrid never gets set to False thus triggering another UUID generation and overwrites self.identifier.